src/Cache/FileSystemCache.php

```diff
@@ -0,0 +1,67 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer\Cache;
+
+use LanguageServer\LanguageClient;
+use Sabre\Event\Promise;
+
+/**
+ * Caches content on the file system
+ */
+class FileSystemCache implements Cache
+{
+    /**
+     * @var string
+     */
+    public $cacheDir;
+
+    public function __construct()
+    {
+        if (PHP_OS === 'Windows') {
```

Comment:
Don't use PHP_OS here, since it is just the system php was built from, not the system it is running at the moment. Use PHP_EOL instead(`\r\n` is typical for Windows)